### DISCUSSION TITLE
Is it possible to add $CURRENT_ACCOUNTABILITY?

### DISCUSSION THREAD

**Discussion by tspvivek** [directus/directus#12534] [Feature Requests]
As there is authenticate hook, with which we can control accountability object. Can there be a $CURRENT_ACCOUNTABILITY dynamic variable which can be used in permissions to make the directus more customizable?

**Comment by rijkvanzanten** [comment]
What would you use that for? $CURRENT_ROLE / $CURRENT_USER is already pulled from accountability currently 🤔

**Reply by tspvivek** [reply]
I am using it like this https://github.com/directus/directus/discussions/9682#discussioncomment-2478210, this is one of the use case. We have authentication hook and option to customise accountability object, we can solve so many use cases with $CURRENT_ACCOUNTABILITY.

**Reply by rijkvanzanten** [reply]
Ahh I see, so you added custom fields to accountability, that you then want to read through things like `$CURRENT_ACCOUNTABILITY.organization` etc

**Reply by tspvivek** [reply]
Yes, this one of my use case but others can also solve many such use cases with it. $CURRENT_USER and $CURRENT_ROLE object needs columns (or relations) in DB, but accountability object is completely dynamic so we can configure it to our will.

**Comment by u12206050** [comment]
We have a use case for Tenants, which arguably could be called organization as well :)
Would still need to find a way to have a type of tenant switch within the frontend directus app, and send tenant id with every request.

**Reply by tspvivek** [reply]
I am using NextJS app as frontend with [Refine & Directus dataprovider](https://github.com/tspvivek/refine-directus), For directus app, we should wait for directus team. :)

**Comment by abdonrd** [comment]
We have been using [patch-package](https://www.npmjs.com/package/patch-package) for a long time to achieve this behavior.

It would be great to be able to have him in the Core of Directus. And I think it makes a lot of sense since the `authenticate` hook allow us to change the accountability.

**Reply by tspvivek** [reply]
Seems core team is busy with sockets and JSON query integrations. Too many changes happening rapidly. I am planning to create a pull request after socket is live.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
